lib/foreman/force_ssl.rb

```diff
@@ -0,0 +1,43 @@
+module Foreman
+  class ForceSsl
+    UNATTENTED_PATHS = %r{^/(unattended|userdata)/}
+
+    def self.allowed_http_regex
+      @allowed_http_regex ||= Regexp.union
+    end
+
+    def self.add_allowed_http_path_regex!(regex)
+      @allowed_http_regex = Regexp.union(allowed_http_regex, regex)
+    end
```

Comment:
I'm still wondering about this. Given we currently don't have a use for it, can we optimize it so it doesn't always use the regex engine? That probably means making `path_allows_http?` and `add_allowed_http_path_regex` deal with `@allowed_http_regex` being `nil`.

Reply:
sure, if we drop the internal usage in #9020, I will simplify this :)